packages/react-vapor/src/components/filepicker/Filepicker.tsx

```diff
@@ -1,36 +1,62 @@
 import classNames from 'classnames';
 import * as VaporSVG from 'coveo-styleguide';
 import * as React from 'react';
+import {connect} from 'react-redux';
 
+import {IReactVaporState} from '../../ReactVapor';
+import {IDispatch} from '../../utils';
+import {FileMetadata, FileUtils} from '../../utils/FileUtils';
 import {Svg} from '../svg';
+import {FilepickerActions} from './FilepickerActions';
+import {FilepickerSelectors} from './FilepickerSelectors';
 
 export interface FilepickerProps {
     id: string;
 }
 
-export const Filepicker: React.FunctionComponent<FilepickerProps & React.HTMLProps<HTMLInputElement>> = (props) => {
-    const {placeholder, ...inputProps} = props;
-    const [fileName, setFileName] = React.useState(null);
+const mapStateToProps = (state: IReactVaporState, ownProps: FilepickerProps) => ({
+    selectedFile: FilepickerSelectors.getFileMetadata(state, ownProps),
+    isEmpty: FilepickerSelectors.isEmpty(state, ownProps),
+});
+
+const mapDispatchToProps = (dispatch: IDispatch, {id}: FilepickerProps) => ({
+    addFilepicker: () => dispatch(FilepickerActions.add(id)),
+    setFile: (file: FileMetadata) => dispatch(FilepickerActions.setFile(id, file)),
+    clear: () => dispatch(FilepickerActions.clear(id)),
+});
+
+const FilepickerDisconnected: React.FunctionComponent<
+    FilepickerProps &
+        React.HTMLProps<HTMLInputElement> &
+        ReturnType<typeof mapDispatchToProps> &
+        ReturnType<typeof mapStateToProps>
+> = (props) => {
+    const {addFilepicker, setFile, clear, isEmpty, selectedFile, placeholder, ...inputProps} = props;
     const input = React.useRef<HTMLInputElement>();
     const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-        setFileName(e.target.files[0]?.name ?? null);
+        setFile(FileUtils.serialize(e.target.files[0]) ?? null);
```

Comment:
is it a best practice to use the nullish coalescing operator with null instead of returning null (instead of undefined) in the `FileUtils.serialize()` method (see line 16 in `FileUtils.ts`) ?

Reply:
Yeah good call, I could probably just remove the `null` here, I added the call to serialize without giving too much attention to the nullish coalescing value.